include/boost/typeof/msvc/typeof_impl_decltype.hpp

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2006 Arkadiy Vertleyb
+// Copyright (C) 2017 Daniela Engert
+// Use, modification and distribution is subject to the Boost Software
+// License, Version 1.0. (http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_TYPEOF_MSVC_TYPEOF_IMPL_DECLTYPE_HPP_INCLUDED
+# define BOOST_TYPEOF_MSVC_TYPEOF_IMPL_DECLTYPE_HPP_INCLUDED
+
+#include <boost/type_traits/is_function.hpp> 
+#include <boost/type_traits/remove_reference.hpp> 
+#include <boost/utility/enable_if.hpp>
```

Comment:
Ok, then this applies to three other header files in this library as well. But IMHO this should be addressed by a different pull-request.